### PR TITLE
Fuzz test additions

### PR DESCRIPTION
**Pull request by prat-gpt**
# Description of change
## High level
* Add ERC20Pool fuzz test for kick, take, and settle with deposit.
* Add ERC721Pool fuzz test for add and remove collateral.

### CODE REVIEW

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol [hunk 1]
```diff
@@ -830,3 +833,98 @@ contract ERC20PoolLiquidationsKickTest is ERC20HelperContract {
         );
     }
 }
+
+contract ERC20PoolLiquidationKickFuzzyTest is ERC20FuzzyHelperContract {
+    address internal _lender;
+    address internal _kicker;
+    address[] internal _borrowers;
+    uint256[5] internal _buckets = [2550, 2551, 2552, 2553, 2554];
```
Comment (EdNoepel) [inline tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: Would be nice if number of buckets and start bucket was fuzzed, but I suppose that's covered by invariant testing.

**Review by EdNoepel — APPROVED** [review]
Okay.

File: tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol (hunk 1 above)
Comment (prat-gpt) [reply tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: Yes, you are right it is covered in invariant testing, adding it here as well. [c272d41](https://github.com/ajna-finance/contracts/pull/924/commits/c272d418ff65d3e4e602951efd26aba276033bac)

File: tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol [hunk 2]
```diff
@@ -830,3 +833,99 @@ contract ERC20PoolLiquidationsKickTest is ERC20HelperContract {
         );
     }
 }
+
+contract ERC20PoolLiquidationKickFuzzyTest is ERC20FuzzyHelperContract {
+    address internal _lender;
+    address internal _kicker;
+    address[] internal _borrowers;
```
Comment (ith-harvey) [inline tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol:840]: NIT: new line under this

File: tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol [hunk 3]
```diff
@@ -830,3 +833,99 @@ contract ERC20PoolLiquidationsKickTest is ERC20HelperContract {
...
+    function setUp() external {
+        _startTest();
+        _lender = makeAddr("lender");
+        _kicker = makeAddr("kicker");
+    }
+
+    function testBorrowAndKickFuzzy(uint256 noOfBorrowers, uint256 totalPoolLiquidity, uint256 startingBucket, uint256 noOfBuckets) external tearDown {
+        noOfBorrowers  = bound(noOfBorrowers, 1, 10);
```
Comment (ith-harvey) [inline tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: NIT: align the `=` here

**Review by ith-harvey — APPROVED** [review]
LGTM

File: tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol (hunk 3 above)
Comment (prat-gpt) [reply tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: Fixed in [d1f7a25](https://github.com/ajna-finance/contracts/pull/924/commits/d1f7a25554cbd4f7d13d114b40a825f75cb837d1)

File: tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol (hunk 2 above)
Comment (prat-gpt) [reply tests/forge/unit/ERC20Pool/ERC20PoolLiquidationsKick.t.sol:840]: Fixed in [d1f7a25](https://github.com/ajna-finance/contracts/pull/924/commits/d1f7a25554cbd4f7d13d114b40a825f75cb837d1)

**Pull request merged**